src/main/java/com/information/user/User.java

```diff
@@ -15,6 +15,10 @@ public static User createUser(String name, String birthdate) {
         return new User(null, name, birthdate);
     }
 
+    public static User updateUser(Integer id, String name, String birthdate) {
```

Comment:
インスタンスメソッドのほうがいいかなと思います。
```suggestion
    public User updateUser(Integer id, String name, String birthdate) {
```

```java
User user = userMapper.findById(id);
user.updateUser(name, birthdate);
```

---
このようにインスタンスを生成するための専用のメソッドをstaticで用意することはあります。
```java
public static User createUser(String name, String birthdate)
```
```java
User user = User.createUser(name, birthdate);
```

Reply:
修正途中ですが、一度見ていただきたいです。

`userMapper.findById(id)` を呼び出して検索したものを `user` 変数に割り当てると理解しているのですが、userMapper の部分は新たにオブジェクトが必要ということでしょうか？

![image](https://github.com/kino41/RaiseTech_last/assets/155221768/9bf103dd-4860-4035-8b91-a584b7c5c8ae)

![image](https://github.com/kino41/RaiseTech_last/assets/155221768/0c9da459-5fef-4d64-93dd-4ba8a387dd41)

登録処理の実装時から、インスタンスメソッドの取扱いを調べているのですが使いどころがいまいち掴めずにいます。

## 調べてわかったこと
- インスタンスを作成せずに直接呼び出しが可能である
- クラスのインスタンスに紐づいているため、インスタンスの状態や振る舞いを受け継ぐものである

UserService クラスに以下を記述しても正常にリクエストが通ったのですが、この場合もインスタンスを作成しているためインスタンスメソッドとはなっていないですよね？
しっかり理解できておらず、すみませんがよろしくお願いいたします。
![image](https://github.com/kino41/RaiseTech_last/assets/155221768/00dfa8f7-0537-4081-9164-72c6f49b5cca)

Reply:
ちょっと修正イメージが違いますね！
User.javaにはUserMapperをフィールドに持たせていないのでfindByIdは呼び出すことができません。
UserMapperを使ってfindByIdをよびだすのはServiceで実装しましょう。

Reply:
やはりそうですよね...
すみません、考えに自信を持てずにいたので一度確認させていただきました！

修正を加えましたのでご確認お願いします。
b88739b9726337f586f3ef7a064df1ee2b9fd677

Reply:
いい感じ！
あとは指定したidに該当するuserがない場合例外にしてレスポンスは404にする処理を入れてください！

Reply:
例外処理を実装しました。

42bfe639d3701b3cbd43befd0e4f4dd5f55b2994

## 動作確認
![image](https://github.com/kino41/RaiseTech_last/assets/155221768/13ca7dc0-eb7d-48f9-b509-2ece667bec38)
存在しない id = 8 を指定してリクエストした場合、
- ステータスコード404確認
- カスタマイズしたエラー内容が返ることを確認